rom/dev/tests/rom_integration_tests/test_wdt_activation_and_stoppage.rs

```diff
@@ -107,4 +107,15 @@ fn test_rom_wdt_timeout() {
     .unwrap();
 
     hw.step_until(|m| m.soc_ifc().cptra_fw_error_fatal().read() == WDT_EXPIRED);
+
+    // no mcause if wdt times out
+    assert_eq!(hw.soc_ifc().cptra_fw_extended_error_info().at(0).read(), 0);
+    // no mscause if wdt times out
+    assert_eq!(hw.soc_ifc().cptra_fw_extended_error_info().at(1).read(), 0);
+    // mepc is a memory address so won't be 0
+    assert_ne!(hw.soc_ifc().cptra_fw_extended_error_info().at(2).read(), 0);
+    // return address won't be 0
+    assert_ne!(hw.soc_ifc().cptra_fw_extended_error_info().at(3).read(), 0);
+    // no error_internal_intr_r if wdt times out
+    assert_eq!(hw.soc_ifc().cptra_fw_extended_error_info().at(4).read(), 0);
```

Comment:
That's weird, I would expect on of these bits to be set:

https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.intr_block_rf.error_internal_intr_r#error_wdt_timer2_timeout_sts

Reply:
Ah, and it is on the FPGA:

```
thread 'test_wdt_activation_and_stoppage::test_rom_wdt_timeout' panicked at 'assertion failed: `(left == right)`
  left: `64`,
 right: `0`', rom/dev/tests/rom_integration_tests/test_wdt_activation_and_stoppage.rs:120:5
```

So this is a problem with the emulator.

Reply:
Removed this last line for now to make CI pass. I looked around in the sw-emulator but not sure what's going wrong.

Reply:
Can you add that assert back? Before we merge this I want to make sure all the right bits are getting set (since until they do, there is still a test gap 😁 )

Reply:
I think to fix this, in [these blocks](https://github.com/chipsalliance/caliptra-sw/blob/3df8b615e95ee5372c8df8e1b742056e0c014394/sw-emulator/lib/periph/src/soc_reg.rs#L1027-L1064), we want to set the [error_wdt_timer*_timeout_sts](https://chipsalliance.github.io/caliptra-rtl/main/internal-regs/?p=clp.soc_ifc_reg.intr_block_rf.error_internal_intr_r#error_wdt_timer2_timeout_sts.desc) bits in the `error_internal_intr_r` register.

Today the whole interrupt register block is just an array of `u32`s: https://github.com/chipsalliance/caliptra-sw/blob/3df8b615e95ee5372c8df8e1b742056e0c014394/sw-emulator/lib/periph/src/soc_reg.rs#L556C15-L556C15.

So the cheap way to write these bits would be to find the right index and set the bit in the corresponding if.

Reply:
There is a nuance in implementing this register that the bits are "write-one-to-clear" (`woclr`). So if we implement it in the cheap way I suggested above, that wouldn't actually work.

Reply:
Done